R/load_pop_data.R

```diff
@@ -2,40 +2,184 @@
+                         antigen_isos = NULL,
+                         age = "Age",
+                         id = "index_id",
+                         value = "result",
+                         standardize = TRUE)
 {
   if(file_path %>% substr(1,4) == "http")
   {
     file_path = url(file_path)
-
   }
 
   pop_data =
     file_path %>% readRDS() %>%
     tibble::as_tibble()
 
-  class(pop_data) =
-    c("pop_data", class(pop_data))
+  # set pop_data class
+  attr(pop_data, "class") = c('pop_data', class(pop_data))
 
   if(is.null(antigen_isos))
   {
     antigen_isos = unique(pop_data$antigen_iso)
   } else
   {
     stopifnot(all(is.element(antigen_isos, pop_data$antigen_iso)))
-
   }
 
   attr(pop_data, "antigen_isos") = antigen_isos
 
+  ##### AGE
+  if(age %in% colnames(pop_data))
+  {
+    attr(pop_data, "age_var") <- age
+  } else
+  {
+    # search age variable from pop_data
+    age_var <- pop_data %>%
+      select(tidyselect::matches("age",colnames(pop_data),ignore.case = TRUE) & ends_with("e", ignore.case = TRUE)) %>%
+      names()
+
+    if(length(age_var) > 0)
+    {
+      attr(pop_data, "age_var") <- age_var
+
+      # create warning when using searched age instead of provided age
+      cli::cli_alert_warning('The provided age attribute "{age}" does not exist.
+                        Proceeding to use "{age_var}"')
+
+    } else
+    {
+      cli::cli_abort(
+        '`age = "{age}"` is not a valid input;
+      the age column in not available in the data set')
+    }
+
+  }
+
+  ##### INDEX
+  if(id %in% colnames(pop_data))
+  {
+    attr(pop_data, "id_var") <- id
+
+
+  } else
+  {
+    # search index variable from pop_data (no need to find ID)
+    id_var <- pop_data %>%
+      select(tidyselect::matches("\\w*id\\b")) %>%
+      names()
+
+
+    if(length(id_var) > 0)
+    {
+      attr(pop_data, "id_var") <- id_var
+
+      # create warning when using searched id  instead of provided id
+      cli::cli_alert_warning('The provided id attribute "{id}" does not exist.
+                        Proceeding to use "{id_var}"')
+    } else
+    {
+      cli::cli_abort(
+        '`id = "{id}"` is not a valid input;
+      the id column in not available in the data set')
+    }
+
+  }
+
+  ##### VALUE
+  if(value %in% colnames(pop_data))
+  {
+    attr(pop_data, "value_var") <- value
+  } else
+  {
+    # search value variable from pop_data
+    value_var <- pop_data %>%
+      select(tidyselect::matches("result", ignore.case = TRUE)) %>%
+      names()
+
+    if(length(value_var) > 0)
+    {
+      attr(pop_data, "value_var") <- value_var
+
+      # create warning when using searched age instead of provided age
+      cli::cli_alert_warning('The provided age attribute "{value}" does not exist.
+                        Proceeding to use "{value_var}"')
+    } else
+    {
+      cli::cli_abort(
+        '`value = "{value}"` is not a valid input;
+      the value column in not available in the data set')
+    }
+
+  }
```

Comment:
please decompose out three subfunctions from this section - one for each of the three variables